db/sequence_allocator.go

```diff
@@ -26,19 +26,28 @@ const (
 )
 
 type sequenceAllocator struct {
-	bucket base.Bucket // Bucket whose counter to use
-	mutex  sync.Mutex  // Makes this object thread-safe
-	last   uint64      // Last sequence # assigned
-	max    uint64      // Max sequence # reserved
+	bucket  base.Bucket    // Bucket whose counter to use
+	dbStats *DatabaseStats // For updating per-db stats
+	mutex   sync.Mutex     // Makes this object thread-safe
+	last    uint64         // Last sequence # assigned
+	max     uint64         // Max sequence # reserved
 }
 
-func newSequenceAllocator(bucket base.Bucket) (*sequenceAllocator, error) {
-	s := &sequenceAllocator{bucket: bucket}
+func newSequenceAllocator(bucket base.Bucket, dbStats *DatabaseStats) (*sequenceAllocator, error) {
+
+	if dbStats == nil {
```

Comment:
Is this just to support tests that don't pass in DatabaseStats?  Seems like this code would make more sense in the tests, to avoid case where we silently fail to update the right DatabaseStats.

Reply:
It's a defensive measure to avoid panics in case this function is ever called with a `dbStats` that's `nil`.  Taking another look, I'll change it to return an error in that case.